src/main/java/org/jabref/gui/bibtexextractor/ExtractBibtexDialog.java

```diff
@@ -22,45 +25,66 @@
+    @Inject protected FileUpdateMonitor fileUpdateMonitor;
+    @Inject protected TaskExecutor taskExecutor;
+    @Inject protected UndoManager undoManager;
+    @Inject protected PreferencesService preferencesService;
+
+    @FXML protected TextArea input;
+    private final boolean onlineMode;
+
     @FXML private ButtonType parseButtonType;
-    private BibtexExtractorViewModel viewModel;
-    @Inject private StateManager stateManager;
-    @Inject private DialogService dialogService;
-    @Inject private FileUpdateMonitor fileUpdateMonitor;
-    @Inject private TaskExecutor taskExecutor;
-    @Inject private UndoManager undoManager;
-    @Inject private PreferencesService preferencesService;
 
-    public ExtractBibtexDialog() {
+    public ExtractBibtexDialog(boolean onlineMode) {
+        this.onlineMode = onlineMode;
         ViewLoader.view(this)
+                  .controller(this)
                   .load()
                   .setAsDialogPane(this);
-        this.setTitle(Localization.lang("Plain References Parser"));
-        input.setPromptText(Localization.lang("Please enter the plain references to extract from separated by double empty lines."));
-        input.selectAll();
-
-        buttonParse = (Button) getDialogPane().lookupButton(parseButtonType);
-        buttonParse.setTooltip(new Tooltip((Localization.lang("Starts the extraction and adds the resulting entries to the currently opened database"))));
-        buttonParse.setOnAction(event -> viewModel.startParsing());
-        buttonParse.disableProperty().bind(viewModel.inputTextProperty().isEmpty());
+        if (onlineMode) {
+            this.setTitle(Localization.lang("Plain References Parser (online)"));
+        } else {
+            this.setTitle(Localization.lang("Plain References Parser (offline)"));
+        }
     }
 
     @FXML
     private void initialize() {
         BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
-        this.viewModel = new BibtexExtractorViewModel(
+        BibtexExtractorViewModel viewModel = new BibtexExtractorViewModel(
+                onlineMode,
                 database,
                 dialogService,
                 preferencesService,
                 fileUpdateMonitor,
                 taskExecutor,
                 undoManager,
                 stateManager);
+
         input.textProperty().bindBidirectional(viewModel.inputTextProperty());
+        String clipText = ClipBoardManager.getContents();
+        if (StringUtil.isBlank(clipText)) {
+            input.setPromptText(Localization.lang("Please enter the plain references to extract from separated by double empty lines."));
+        } else {
+            input.setText(clipText);
+            input.selectAll();
+        }
+
+        Platform.runLater(() -> {
+            input.requestFocus();
+            Button buttonParse = (Button) getDialogPane().lookupButton(parseButtonType);
```

Comment:
I need to have this code here, because otherwise, I get

    Caused by: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.Button.setTooltip(javafx.scene.control.Tooltip)" because "buttonParse" is null